server/src/main/java/io/druid/server/coordinator/CostBalancerStrategy.java

```diff
@@ -41,12 +43,24 @@
   private static final long SEVEN_DAYS_IN_MILLIS = 7 * DAY_IN_MILLIS;
   private static final long THIRTY_DAYS_IN_MILLIS = 30 * DAY_IN_MILLIS;
   private final long referenceTimestamp;
-  private final int threadCount;
+  private final ListeningExecutorService exec;
+
+  public static long gapMillis(Interval interval1, Interval interval2)
+  {
+    if (interval1.getStartMillis() > interval2.getEndMillis()) {
+      return interval1.getStartMillis() - interval2.getEndMillis();
+    } else if (interval2.getStartMillis() > interval1.getEndMillis()) {
+      return interval2.getStartMillis() - interval1.getEndMillis();
+    } else {
+      return 0;
+    }
+  }
 
   public CostBalancerStrategy(DateTime referenceTimestamp, int threadCount)
```

Comment:
instead of making strategies closeable, how about we inject the executor service and attach it to the lifecycle? I think it would make things cleaner.